homeassistant/components/group/media_player.py

```diff
@@ -0,0 +1,350 @@
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_MEDIA_SEEK, data, blocking=True, context=self._context,
+        )
+
+    def media_stop(self):
+        """Send stop command."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_PAUSE_PLAY_STOP]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_MEDIA_STOP, data, blocking=True, context=self._context,
+        )
+
+    def mute_volume(self, mute):
+        """Mute the volume."""
+        data = {
+            ATTR_MEDIA_VOLUME_MUTED: mute,
+            ATTR_ENTITY_ID: self._players[KEY_VOLUME],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_VOLUME_MUTE, data, blocking=True, context=self._context,
+        )
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        data = {
+            ATTR_MEDIA_CONTENT_ID: media_id,
+            ATTR_MEDIA_CONTENT_TYPE: media_type,
+            ATTR_ENTITY_ID: self._players[KEY_PLAY_MEDIA],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_PLAY_MEDIA, data, blocking=True, context=self._context,
+        )
+
+    def turn_on(self):
+        """Forward the turn_on command to all media in the media group."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_ON_OFF]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_TURN_ON, data, blocking=True, context=self._context,
+        )
+
+    def set_volume_level(self, volume):
+        """Set volume level(s)."""
+        data = {
+            ATTR_MEDIA_VOLUME_LEVEL: volume,
+            ATTR_ENTITY_ID: self._players[KEY_VOLUME],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_VOLUME_SET, data, blocking=True, context=self._context,
+        )
+
+    def turn_off(self):
+        """Forward the turn_off command to all media in the media group."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_ON_OFF]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_TURN_OFF, data, blocking=True, context=self._context,
+        )
+
+    async def async_volume_up(self):
+        """Turn volume up for media player(s)."""
+        for entity in self._players[KEY_VOLUME]:
+            volume_level = self.hass.states.get(entity).attributes["volume_level"]
+            if volume_level < 1:
+                await self.async_set_volume_level(min(1, volume_level + 0.1))
+
+    async def async_volume_down(self):
+        """Turn volume down for media player(s)."""
+        for entity in self._players[KEY_VOLUME]:
+            volume_level = self.hass.states.get(entity).attributes["volume_level"]
+            if volume_level > 0:
+                await self.async_set_volume_level(max(0, volume_level - 0.1))
+
+    async def async_update(self):
+        """Query all members and determine the media group state."""
+        all_states = [self.hass.states.get(x) for x in self._entities]
+        not_none_states: List[State] = list(filter(None, all_states))
```

Comment:
Please use a list comprehension instead of filter.

Reply:
Done